src/Compositor.cpp

```diff
@@ -2531,12 +2532,13 @@ Vector2D CCompositor::parseWindowVectorArgsRelative(const std::string& args, con
     bool        yIsPercent = false;
     bool        isExact    = false;
 
-    std::string x = args.substr(0, args.find_first_of(' '));
-    std::string y = args.substr(args.find_last_of(' ') + 1);
+    CVarList    varList(args, 3, 's', true);
+    std::string x = varList.size() > 0 ? varList[0] : "";
+    std::string y = varList.size() > 1 ? varList[1] : "";
 
     if (x == "exact") {
-        x       = y.substr(0, y.find_first_of(' '));
-        y       = y.substr(y.find_first_of(' ') + 1);
+        x       = varList.size() > 1 ? varList[1] : "";
```

Comment:
no need to check size, CVarList will return `""` if it's outside of its size

Reply:
Excuse me, if I understand you correctly, is it safe to remove all four size checks in the PR ?

Also, I have tested this and tabs characters are not working even when using 's' as delim.
Thanks for your patience.

Reply:
are you testing sith `\t`? you need actual tab, which may appear in config
ctrv+v and press tab if u wish from cmdline. or insert
```sh
hyprctl dispatch resizeactive exact 80%		40%
```

Reply:
Yes, I have inserted an actual tab character in different editors. Also, if you want to test it, notice that in the line of code that you wrote there is a space before the tab.

Reply:
Thank you, now that you tell, I have realized where is the error. It only works with a tab character when using the `exact` argument.

Reply:
hyprctl dispatch resizeactive exact 80%\t40%    = fail
hyprctl dispatch resizeactive exact\t80%\t40%  = fail
hyprctl dispatch resizeactive exact\t80% 40%    = success
hyprctl dispatch resizeactive exact 80%\t40%    = success

Reply:
maybe `std::isspace` doesn't catch a tab? tho it should I believe.

Anyways, `is it safe to remove all four size checks in the PR ?`

si senior.

Reply:
yeah I tested it, it seem like you need to set `0` instead of `3` maxsize
I forgot I had 0 here locally

```ini
bind = SUPER CTRL, Z, resizeactive,     	    exact    	      60%		40%  56
```
works, which is mix of spaces and tabs


about `\t`, it's not expanded to real tab when you run like that.
if you want to test it from hyprctl
```sh
hyprctl dispatch resizeactive $(echo -e "exact\t60%\t40%")
```

Reply:
It catches the tab, but the arguments also need to contain a space for some reason.

Reply:
Yea, I wrote \t for it to be clear that they were actual tabs.
Can you try with one of these, replacing <tab> with an actual tab ?
hyprctl dispatch resizeactive 80%<tab>40%
hyprctl dispatch resizeactive exact<tab>80%<tab>40%

It works when you mix tabs and spaces, but not when you only write tabs. Look at this, I hope it is clear now:

Works:
+ Arg1<tab>Arg2<space>Arg3
+ Arg1<space>Arg2<tab>Arg3

Fails:
-  Arg1<tab>Arg2<tab>Arg3
 
Thanks.

Reply:
> yeah I tested it, it seem like you need to set 0 instead of 3 maxsize

or `4` if you wish, that makes no difference
dunno why it returns wrong when 3
```cpp
    Debug::log(WARN, "exact:{}  x:@{}@  y:@{}@",  isExact, x, y);
    Debug::log(WARN, "size:{}", varList.size());
    Debug::log(WARN, "1:{}", varList[0]);
    Debug::log(WARN, "2:{}", varList[1]);
    Debug::log(WARN, "3:{}", varList[2]);
```

```
[WARN] exact:true  x:@80%@  y:@80%	 	40%@
[WARN] size:3
[WARN] 1:exact
[WARN] 2:80%
[WARN] 3:80%	 	40%
```
with `0` or `4`
```
[WARN] exact:true  x:@80%@  y:@40%@
[WARN] size:3
[WARN] 1:exact
[WARN] 2:80%
[WARN] 3:40%
```

Reply:
I have compiled it with `0` and `4` with the same results as with `3`

Reply:
ohmygod sure xD
look at this [line](https://github.com/hyprwm/Hyprland/pull/5262/files#diff-3c089c6c6b23a85ab5b4cffb002e8880ad069c8a8b3462f7aafa9e5345c0333dR2526)

just do
```cpp
if (!args.contains(' ') && !args.contains('\t'))
    return relativeTo;
```

and with `0` maxsize all stuff works, spaces, tabs, mixed, hyprctl or conf

Reply:
Actually, if I use Control+V+Tab to get a tab character in the terminal as @bvr-yr suggested, it works perfectly fine in every situation.

However, it has problems with tab characters from GUI editors like VSCode, kate, or gedit.

If someone else can test this it would be good. Thanks.

Reply:
I'm not sure what's wrong with 3 maxsize, but it returns strange things like this
```
[WARN] exact:true  x:@60%@  y:@0%  40%@
[WARN] size:3
[WARN] 1:exact
[WARN] 2:60%
[WARN] 3:0%  40%
[ERR] parseWindowVectorArgsRelative: args not numbers
```
for example with bind
```ini
bind = SUPER CTRL, Z, resizeactive, exact    60%  40%
```
anyways, initial fix was config issues, so let's see what boss will tell about
```cpp
if (!args.contains(' ') && !args.contains('\t'))
    return relativeTo;
```
and 0 maxsize

Reply:
tag me when this is ready. And yes, generally 0 maxsize is used